src/DynamoCore/Core/PathManager.cs

```diff
@@ -201,7 +201,10 @@ internal PathManager(PathManagerParams pathManagerParams)
             }
 
             // Current user specific directories.
-            userDataDir = GetUserDataFolder();
+            if (pathResolver == null || pathResolver.UserDataRootFolder == null)
+                userDataDir = GetUserDataFolder();
+            else
+                userDataDir = pathResolver.UserDataRootFolder;
```

Comment:
Hi @nguyen-binh-minh, please pass `pathResolver` into `GetUserDataFolder` and fold this logic in there so this line of code is not changed. Do the same for `GetCommonDataFolder`. Thanks!